components/sup/src/manager/self_updater.rs

```diff
@@ -31,8 +31,13 @@ pub const SUP_PKG_IDENT: &'static str = "core/hab-sup";
 const DEFAULT_FREQUENCY: i64 = 60_000;
 const FREQUENCY_ENVVAR: &'static str = "HAB_SUP_UPDATE_MS";
 
+enum UpdateStrategy {
```

Comment:
I really like this, except for the name... `UpdateStrategy` already has a history in the Supervisor as a completely separate thing, and this could get confusing.

Maybe something like `PollingStrategy`? I'm not quite sure of a great name; I'll think on it a bit.

Reply:
How about just `Strategy`? This is private to `self_updater` module, so I don't think that should cause confusion.

Reply:
Good point.

![tenor-89596809](https://user-images.githubusercontent.com/207178/43661320-c9e3a2fe-972f-11e8-9726-918a59b1e0f1.gif)